src/rust-crypto/RoomEncryptor.ts

```diff
@@ -223,53 +264,20 @@ export class RoomEncryptor {
         });
     }
 
-    /**
-     *  The Rust-SDK requires that we only have one shareRoomKey process in flight at once for a room.
-     *  This method ensures that, by only having one call to shareRoomKey active at once (and making them
-     *  queue up in order).
-     *
-     * @param userList - list of userIDs to share with
-     * @param rustEncryptionSettings - encryption settings to use
-     *
-     * @returns a promise which resolves to the list of ToDeviceRequests to send
-     */
-    private async shareRoomKey(
-        userList: UserId[],
-        rustEncryptionSettings: EncryptionSettings,
-    ): Promise<ToDeviceRequest[]> {
-        const prom = this.currentShareRoomKeyPromise
-            .catch(() => {
-                // any errors in the previous claim will have been reported already, so there is nothing to do here.
-                // we just throw away the error and start anew.
-            })
-            .then(() => this.olmMachine.shareRoomKey(new RoomId(this.room.roomId), userList, rustEncryptionSettings));
-        this.currentShareRoomKeyPromise = prom;
-        return prom;
-    }
-
     /**
      * Discard any existing group session for this room
      */
     public async forceDiscardSession(): Promise<void> {
+        // XXX Clarify if also need to ensure order for that call to.
+        // It will mark the session as invalidated but not delete it, so if the current encryption promise
+        // has already check for rotation it will not be affected.
```

Comment:
Pretty sure it doesn't matter.

This is only called if someone does `/discardsession` in Element. If you do that while there are messages queued, then yes, there's a possibility that some of the messages in the queue will go to the old session. But that's ok?